modules/identity-provider-add.adoc

```diff
@@ -37,15 +48,50 @@ $ oc apply -f </path/to/CR>
 If a CR does not exist, `oc apply` creates a new CR and might trigger the following warning: `Warning: oc apply should be used on resources created by either oc create --save-config or oc apply`. In this case you can safely ignore this warning.
 ====
 
+ifndef::no-username-password-login[]
 . Log in to the cluster as a user from your identity provider, entering the
 password when prompted.
 +
 ----
 $ oc login -u <username>
 ----
+endif::no-username-password-login[]
+
+ifdef::no-username-password-login[]
+
+. Obtain a token from the OAuth server.
+
+.. You can request tokens at `<namespace_route>/oauth/token/request`. The value of `<namespace_route>` can be found using the following command:
++
+----
+$ oc get route oauth-openshift -n openshift-authentication -o json | jq .spec.host
+----
+
+. Log in to the cluster, passing in the token to authenticate.
```

Comment:
It might be worth documenting that you're going to be prompted to log in on the above URL, the command for `oc` login should appear there. Please do go ahead and try that in your cluster :)

Reply:
When would they be prompted to log in with a URL?